feat(msgpack_rpc): add a new `msgpack-rpc` client type to fix behavior with MessagePack-RPC compliant clients (#24399)
neovim/neovim: runtime/doc/api.txt

      • {type}        Must be one of the following values. Client libraries
                      should default to "remote" unless overridden by the
                      user.

>                       • "remote" remote client connected "Nvim flavored"
>                         MessagePack-RPC (responses must be in reverse order of
>                         requests).

Reviewer: ```suggestion
                      • "remote" remote client connected "Nvim flavored"
                        MessagePack-RPC (responses must be in reverse order of
                        requests). |msgpack-rpc|
```

      • {type}        Must be one of the following values. Client libraries
                      should default to "remote" unless overridden by the
                      user.

>                       • "remote" remote client connected to Nvim via Neovim
>                         modified MessagePack-RPC that is not fully
>                         MessagePack-RPC compliant (out-of-ordered responses
>                         are not supported).

Reviewer: I'm working on updating `:help msgpack-rpc` but let's link to it meanwhile.
```suggestion
                      • "remote" remote client connected "Nvim flavored"
                        MessagePack-RPC (responses must be in reverse order of
                        requests).
```

                      • "msgpack-rpc" remote client connected to Nvim via

>                         fully MessagePack-RPC compliant protocol.

Reviewer: This seems like an ok compromise for now. In the future we can revisit `nvim_chan_set` if needed.
